sqlglot/optimizer/qualify_tables.py

```diff
@@ -114,5 +114,14 @@ def _qualify(table: exp.Table) -> None:
                 if isinstance(udtf, exp.Values) and not table_alias.columns:
                     for i, e in enumerate(udtf.expressions[0].expressions):
                         table_alias.append("columns", exp.to_identifier(f"_col_{i}"))
+            else:
+                for node, parent, _ in scope.walk():
+                    if (
+                        isinstance(node, exp.Table)
+                        and not node.alias
+                        and isinstance(parent, (exp.From, exp.Join))
```

Comment:
when does the parent matter? could you just do find_all?

Reply:
It matters because you can have `Table`s in `CREATE TABLE <table>` statements, as well as `INSERT INTO` statements and also hints. I initially did the find_all version but had several test failures because of this. For example:

```diff
- WITH cte AS (SELECT _q_0.a AS a FROM (SELECT x.a AS a FROM x AS x) AS _q_0) CREATE TABLE s AS SELECT cte.a AS a FROM cte AS cte
+ WITH cte AS (SELECT _q_0.a AS a FROM (SELECT x.a AS a FROM x AS x) AS _q_0) CREATE TABLE s AS s AS SELECT cte.a AS a FROM cte AS cte
?                                                                                              +++++
```

```diff
- WITH cte AS (SELECT y.b AS b FROM y AS y) INSERT INTO s SELECT cte.b AS b FROM cte AS cte
+ WITH cte AS (SELECT y.b AS b FROM y AS y) INSERT INTO s AS s SELECT cte.b AS b FROM cte AS cte
?                                                        +++++
```

```diff
- SELECT /*+ BROADCAST(x), MERGE(x, y) */ x.a AS a, y.c AS c FROM x AS x JOIN y AS y ON x.b = y.b
+ SELECT /*+ BROADCAST(x AS m), MERGE(x AS m, y AS n) */ x.a AS a, y.c AS c FROM x AS x JOIN y AS y ON x.b = y.b
?                       +++++          +++++   +++++
```